bob/src/cluster/quorum.rs

```diff
@@ -318,6 +319,93 @@ impl Quorum {
         }
         Ok(())
     }
+
+    async fn collect_remote_exists(
+        result: &mut [bool],
+        keys: &[BobKey],
+        indexes_by_node: &mut HashMap<Node, IndexMap>,
+    ) {
+        if indexes_by_node.is_empty() {
+            return;
+        }
+
+        let mut node_keys_by_node_name = HashMap::new();
+        for (node, node_map) in indexes_by_node.iter_mut() {
+            node_map.retain_not_existed(&result);
+            if !node_map.is_empty() {
+                node_keys_by_node_name.insert(node.name().to_owned(), (node.clone(), node_map.collect(keys)));
+            }
+        }
+
+        if !node_keys_by_node_name.is_empty() {
+            let nodes: Vec<_> = node_keys_by_node_name.values().map(|(n, _)| n).collect();
+            let remote_results = exist_on_remote_nodes(&nodes, &node_keys_by_node_name).await;
+            for remote_result in remote_results.into_iter() {
+                match remote_result {
+                    Ok(remote_result) => {
+                        let node = &node_keys_by_node_name.get(remote_result.node_name()).unwrap().0;
```

Comment:
Use `expect()` instead of `unwrap()`. `expect` allows to pass a message, describing the problem